src/table/hooks/use-snapshot.ts

```diff
@@ -13,6 +14,12 @@ interface UseSnapshotProps {
 }
 
 const useSnapshot = ({ layout, viewService, model, rootElement, contentRect }: UseSnapshotProps) => {
+  const totalRowCount = layout.qHyperCube.qSize.qcy;
+  const visualRowsPerPage = viewService.rowsPerPage || initialPageInfo.rowsPerPage;
+  const getVisibleHeight = (visibleRowEndIndex: number, visibleRowStartIndex: number) => {
+    return Math.min(totalRowCount, visualRowsPerPage, visibleRowEndIndex - visibleRowStartIndex + 4);
+  };
```

Comment:
no real point of doing this as a function, since you will always use a `visibleHeight` value. I'd suggest something like this, baking in the `visibleRowEndIndex < 0`
```suggestion
  let visibleHeight = 0
  if (visibleRowEndIndex >= 0) {
    const totalRowCount = layout.qHyperCube.qSize.qcy;
    const rowsPerPage = viewService.rowsPerPage || initialPageInfo.rowsPerPage;
    // Adding three extra rows to make sure will fill the space from the pagination
    const extendedVisibleHeight = visibleRowEndIndex - visibleRowStartIndex + 4;
    visibleHeight = Math.min(totalRowCount, visualRowsPerPage, extendedVisibleHeight)
  }
  }
```

Reply:
Fixed using [this](https://github.com/qlik-oss/sn-table/pull/1035/commits/35a0a95f0285cc1df753900875b1077f1df07dc1).